crates/ra_assists/src/assists/add_missing_impl_members.rs

```diff
@@ -227,6 +237,47 @@ fn substitute_type_params<N: AstNode>(
     }
 }
 
+use hir::PathResolution;
+
+// TODO handle generic args
+// TODO handle associated item paths
+// TODO handle value ns?
+
+// FIXME extract this to a general utility as well
```

Comment:
👍

Reply:
@matklad Where do you think utilities like this should go? Since we need it in `ra_assists`, it can't be in `ra_ide`, so the only possibilities seem to be `ra_assists` (but publicly exported), `ra_hir` or a new crate...

Reply:
I think something like `ra_assists/utils` makes sense right now! Long term, maaaybe we need something like this:

```
ra_ide_db  -- defines concrete salsa database, including symbol search, and some core IDE utils (like find reference)
   ^
   |
ra_assits -- defines assits
   ^
   |
ra_ide  -- defines Analysis API and the rest of the features. 
```

Reply:
Maybe it even makes sense to do `ra_ide_db` sooner rather than later? Earlier monomorphisation of database may help with compile times.

Reply:
Wouldn't it also help for auto imports?

Reply:
it would, and in fact our import path already has some public function which is used elsewhere

Reply:
Then it would be really great to have an initial `ra_ide_db` version soonish, where more auto import supplementary functionality is available.

I can help to come up with somehting, but I'm afraid it might cause some lengthy api discussions on the PR 😃

Reply:
I think for autoimport, it will be enough to have it as a public function in `ra_assists`.